packages/core/src/transport-http.ts

```diff
@@ -597,7 +603,7 @@ export class StreamableHttpTransport {
       }
     }
 
-    this.sessions.delete(sessionId);
+    await Promise.resolve(this.sessionStore?.delete(sessionId));
```

Comment:
we don't need the Promise.resolve wrapper here - await should normalize the sync and async return values